src/sentry/grouping/ingest/seer.py

```diff
@@ -176,9 +178,26 @@ def _circuit_breaker_broken(event: Event, project: Project) -> bool:
     return circuit_broken
 
 
+def _has_empty_stacktrace_string(event: Event, variants: Mapping[str, BaseVariant]) -> bool:
+    stacktrace_string = get_stacktrace_string(get_grouping_info_from_variants(variants))
+    # store the stacktrace string in the event so we only calculate it once and so that
+    # we only calculate it if none of the previous checks in should_call_seer_for_grouping fail
```

Comment:
Nit: I'd split this comment in two, since only half of it is actually the reason we're storing the string on the event. 

Here:

```suggestion
    # Store the stacktrace string in the event so we only calculate it once. (We don't need to worry about it being
    # accidentally stored in the database because only the data in `event.data.data` gets stored.)
```

And before the call to this function, something along the lines of:

```python
    # The rate limit check has to be last (see below) but rate-limiting aside, call this after other checks
    # because it calculates the stacktrace string, which we only want to spend the time to do if we already
    # know the other checks have passed.
```